### PR TITLE
Create Makefile for Converting .bdf to .c

### PR DESCRIPTION
**Pull request by Kfeavel**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Ignore build directory
+build/**
+# macOS metadata
+.DS_Store
```
Comment (micahswitzer) [inline .gitignore:4]: Is there any reason you can't put this in your global gitignore since it's not specific to this project?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+.PHONY: all
+all: u_vga16.bdf
+
+# Panix kernel build target
+.PHONY: panix
+panix: u_vga16.bdf
+
+# bdf2c binary
+.PHONY: bdf2c
+bdf2c:
```
Comment (micahswitzer) [inline Makefile:60]: And reason why this is a phony target? The `%.bdf` target depends on it, so why not let make do it's thing and automatically build the binary when necessary?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+export BDF_TO_C        = $(addsuffix .c, $(basename $@))
+export BUILD_DIRS_MAKE = $(shell mkdir -p $(BUILD_DIR))
+
+# *****************
+# * Build Targets *
+# *****************
+
+# Build all fonts listed
+.PHONY: all
+all: u_vga16.bdf
```
Comment (micahswitzer) [inline Makefile:52]: I feel like this should have `panix` as a prerequisite, unless this target is disjoint from the `panix` one.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,24 @@
-# fonts
-Fonts for the Panix kernel
+# Panix Fonts
+
+Font source and source creation tools for the Panix kernel. This repo is *not* submoduled by the main Panix kernel repo and instead is an auxilialry helper repo used to generate VGA fonts to be included by the kernel. This is subject to change in the future if / when Panix gains support for loading VGA / PSF fonts.
```
Comment (micahswitzer) [inline README.md:3]: So how do the c files make it into the kernel?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+# *****************************
+# * Source Code & Directories *
+# *****************************
+
+# Directories & files
+export ROOT           := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+export BDF2C_DIR      := $(ROOT)/bdf2c-code
+export BUILD_DIR      := $(ROOT)/build
+# Update path for local binaries
```
Comment (micahswitzer) [inline Makefile:39]: That's not how `PATH` works, the path environment variable contains *absolute paths* to folders containing binaries, not the binaries themselves. Regardless, that binary is not in the root folder and it is invoked by a direct reference, so this line seems unnecessary to begin with.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+export COLOR_COM  := \033[0;34m
+export COLOR_OK   := \033[0;32m
+export COLOR_INFO := \033[0;93m
+export COLOR_NONE := \033[m
+
+# *************
+# * Toolchain *
+# *************
+BDF2C := bdf2c
```
Comment (micahswitzer) [inline Makefile:29]: This variable is only used on the `PATH` line below. Either use it everywhere, or remove it.

File: .gitignore (hunk 1 above)
Comment (Kfeavel) [reply .gitignore:4]: True, I always forget about that.

File: Makefile (hunk 3 above)
Comment (Kfeavel) [reply Makefile:52]: It's disjoint from the panix one. `panix` is "build the font selected for the panix kernel" whereas `all` is "build all fonts we place in this repo"

File: README.md (hunk 4 above)
Comment (Kfeavel) [reply README.md:3]: Added manually. This is just supposed to be a tool to help generate the source.

File: Makefile (hunk 3 above)
Comment (micahswitzer) [reply Makefile:50]: ahhh okay

File: README.md (hunk 4 above)
Comment (micahswitzer) [reply README.md:3]: Interesting, so we're going to copy-paste the c files into the panix repo and commit them?

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# Ignore build directory
+build/**
```
Comment (micahswitzer) [inline .gitignore:2]: Sorry I didn't catch this earlier, but you don't need a recursive glob here. You can simply specify the directory.

**Review by micahswitzer — APPROVED** [review]

**Pull request merged**